plugins/search-backend-module-api-docs/src/search/spec-parsers/OpenAPISpecParser.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { major } from 'semver';
+import { SpecParser } from './SpecHandler';
+import { OpenAPI, OpenAPIV3, OpenAPIV2 } from 'openapi-types';
+import { parse } from 'yaml';
+
+/** @public */
+export class OpenAPISpecParser implements SpecParser {
```

Comment:
Yeah, it's easy to imagine this evolving to have many implementations and uses outside of just the search collator.  But perhaps it's best to start small and keep it in here?  Multiple specs could evolve in-place, and then if/when there's a use for them outside of this package, they could be broken out.

Maybe it'd be worth marking this internal instead of public, even?